third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -121,6 +121,12 @@ func resourceSqlDatabaseInstance() *schema.Resource {
 							Computed:    true,
 							Description: `Used to make sure changes to the settings block are atomic.`,
 						},
+						"deletion_protection": {
+							Type:        schema.TypeBool,
+							Default:     false,
```

Comment:
How useful is this field with a `false` default? The reason why we added it to bigtable was to prevent people from accidentally destroying their instances because they didn't realize they should do a lifecycle prevent_destroy. If the default is false, wouldn't that just be equivalent to the prevent_destroy behavior?

Reply:
I didn't consider that. I assumed that the users would set this true if they wanted to avoid terraform deleting their data. I never understood how the OP in 7072 blindly ran a plan that marked the instance for destruction and asked for more guardrails.

I'll set it to true.

Reply:
In fairness, Google doesn't stop the instance from being deleted if a database already exists on it which is strange considering Google requires depedant resource to be deleted before the parent resource is deleted. ie DNS Records in DNS Zones, VPC Networks, GCE Disk, etc